src/config.ml

```diff
@@ -240,9 +241,13 @@ module Conf_map = struct
       (fun err -> `Msg ("not a valid config: " ^ err))
       ( ensure_mem Cipher "config must specify 'cipher AES-256-CBC'"
       >>= fun () ->
-        if mem Cipher t && get Cipher t <> "AES-256-CBC" then
-          Error "currently only supported Cipher is 'AES-256-CBC'"
-        else Ok () )
+        (if mem Cipher t && get Cipher t <> "AES-256-CBC" then
+           Error "currently only supported Cipher is 'AES-256-CBC'"
+         else Ok ())
+        >>= fun () ->
+        match find Auth t with
+        | None | Some `SHA1 -> Ok ()
+        | Some _h -> Error "currently only supported auth is 'SHA1'" )
```

Comment:
to me it looks like you've adjusted all the parts where `SHA1` was hardcoded... and my current guess it, any other (supported) hmac just works. did you try that?

Reply:
I have not tried that yet. With main branch I get errors on the openvpn 2.6 server such as `TLS Error: unknown opcode received from [AF_INET]192.168.1.84:58614 op=12` and a few of `TLS Error: client->client or server->server connection attempted from [AF_INET]192.168.1.84:58614`. It seems to get stuck after the handshake. I should test this because if I can get as far with a different `auth` we exercise the code path and it's no worse.

Reply:
oh, even after #98 you're not able to establish a connection?

Reply:
With the other issues fixed I tried with `auth sha256`, and that does not seem to work: `TLS Error: cannot locate HMAC in incoming packet from [AF_INET]192.168.1.84:37694`

Reply:
It was due to an optional argument with default value `` `SHA1`` that was never passed.